src/EditorFeatures/Core/Shared/Utilities/ThreadingContext.cs

```diff
@@ -49,5 +53,30 @@ public JoinableTaskFactory JoinableTaskFactory
         {
             get;
         }
+
+        public JoinableTaskCollection ShutdownBlockingTasks { get; }
```

Comment:
is this a mutable collection?  i would prefer we expose an entrypoint like `void AddShutdownBlockingTask(TAsk)`.  That way consumers can only add to this, and only this type can effectively do things like join on it.

Reply:
For now I took the approach of `AsyncPackage`.